types/qs/index.d.ts

```diff
@@ -45,6 +46,9 @@ declare namespace QueryString {
         charset?: "utf-8" | "iso-8859-1" | undefined;
         charsetSentinel?: boolean | undefined;
         interpretNumericEntities?: boolean | undefined;
+        allowEmptyArrays?: boolean | undefined;
+        decodeDotInKeys?: boolean | undefined;
```

Comment:
this doesn’t encapsulate that allowDots being false and decodeDotInKeys being true is a “never” scenario. I’d like the types to reflect that if possible.

Reply:
Hello. Exactly what kind of revision do you want?

Reply:
I’d like the types to be such that it’s a type error if i pass `{ allowDots: false, decodeDotInKeys: true }`

Reply:
I made some corrections. **"encodeDotInKeys"**, **"decodeDotInKeys"** parameters have been changed depending on the generic type.

<img width="610" alt="ss" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/36919703/b87c5922-4136-4976-924d-ea9690554d37">

Reply:
awesome, thanks! it'd be great to add ExpectError tests to cover that, otherwise LGTM :-)

Reply:
When I made the changes, the tests of the qs-middleware package were failing. I added another fix for this.